app/models/shipit/task.rb

```diff
@@ -340,7 +340,8 @@ def emit_hooks
     end
 
     def hook_event
-      self.class.name.demodulize.underscore.to_sym
+      event = self.class.name.demodulize.underscore.to_sym
+      append_retry_if_necessary(event)
```

Comment:
I think [`task_serializer`](https://github.com/Shopify/shipit-engine/blob/dd5fb153d14be67c3ad12adc7a9c86b9c454bbc3/app/serializers/shipit/task_serializer.rb#L9) or `deploy_serializer` is used for these webhooks, so could we instead add `retry_attempt` to the serializer's attribute fields and avoid the need for a different event type and this wrapper function?

Then the Spy end should only need to check that `retry_attempt > 0`.

Reply:
Good point, updated to remove the `append_retry_if_necessary` method and added the `max_retries` and `retry_attempt` values to the `task_serializer`